ai_agent_apigen/fastapi_agent_scaffold/main.py

```diff
@@ -113,6 +123,34 @@ async def run_ga():
     except Exception as e:
         return {"status": "error", "message": str(e)}
 
+@app.post("/api/allocate")
+async def allocate(payload: AllocationPayload):
+    """
+    1) Save SNA multipliers for GNN
+    2) Execute GNN (reads sna_weights internally)
+    3) Execute GA (optimises GNN output)
+    """
+    try:
+        # persist the weights for the GNN to consume
+        db["sna_weights"].delete_many({})
+        db["sna_weights"].insert_one(payload.dict())
```

Comment:
You could probably maintain DRY by calling the weight changing route's function instead